### ISSUE TITLE
[ECS, CloudFormation, SecretsManager] [Feature request]: Hide secrets from appearing in the task definition

### ISSUE THREAD

**Issue by kauphylover** [aws/containers-roadmap#250]
**Tell us about your request**
There's CFN integration available with SecretsManager. This is nice but the secrets currently show up on the ECS console under Task definitions. This can be a deal breaker for some users. This is a request to improve the ECS console to hide the secrets by default and have an option "Get/Retrieve Environment variables (secrets in this case)" or some other mechanism to hide these secrets from the console.

**Which service(s) is this request for?**
Fargate, ECS

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
Improve the security posture on the CFN-ECS-SecretsManager integration

**Are you currently working around this issue?**
No

**Additional context**

**Attachments**

**Comment by michaelwittig** [comment]
Is this related to #97 ? You can pass secrets into containers without exposing them. Only CloudFormation does not support it https://docs.aws.amazon.com/AmazonECS/latest/developerguide/specifying-sensitive-data.html

**Comment by kauphylover** [comment]
@michaelwittig Related, yes but this feature request is, until #97 is resolved, valid only for the CFN integration with ECS and SecretsManager. I suspect that when #97 is implemented, it will have the same drawback described in this issue.

**Comment by clareliguori** [comment]
@kauphylover In the task definition "secrets" section, the contents of the secret are never shown (unlike the "environment" section).  Only the ARN of the Parameter Store parameter or Secrets Manager secret are shown.
```
"secrets": [
    {
        "name": "environment_variable_name",
        "valueFrom": "arn:aws:ssm:region:aws_account_id:parameter/parameter_name"
    }
]
```

With #97 (CFN support for secrets), it would be the same - only the ARN (or resource reference) would be shown, something like:
```
Secrets:
- Name: environment_variable_name
  Value: arn:aws:ssm:region:aws_account_id:parameter/parameter_name
```

Does that address your concern?

**Comment by kauphylover** [comment]
@clareliguori Thanks so much. I wasn't aware of a separate section called 'secrets' in the ECS task definition (I've only used CFN, never the ECS CLI or API directly!). Yes, #97 does address my concern (@michaelwittig - I now understand what you meant now, thanks!). My initial reading of #97 gave me the impression that it was to support the param store's SecretString type through CFN through the env variables section.

Closing this since this will be resolved by #97.
